test/performance/benchmarks/scale-from-zero/continuous/main.go

```diff
@@ -0,0 +1,210 @@
+	"knative.dev/serving/test/performance"
+	"log"
+	"sync"
+	"testing"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+
+	"knative.dev/pkg/injection/sharedmain"
+	pkgTest "knative.dev/pkg/test"
+	"knative.dev/serving/pkg/apis/autoscaling"
+	ktest "knative.dev/serving/pkg/testing/v1alpha1"
+	"knative.dev/serving/test"
+	v1a1test "knative.dev/serving/test/v1alpha1"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+var (
+	parallelCount   = flag.Int("parallel", 0, "The count of ksvcs we want to run scale-from-zero in parallel")
+)
+
+const (
+	testNamespace            = "default"
+	serviceName              = "perftest-scalefromzero"
+	helloWorldExpectedOutput = "Hello World!"
+	helloWorldImage          = "helloworld"
+	waitToServe              = 10 * time.Minute
+)
+
+func clientsFromConfig() (*test.Clients, error) {
+	cfg, err := sharedmain.GetConfig("", "")
+	if err != nil {
+		return nil, fmt.Errorf("error building kubeconfig: %v", err)
+	}
+	return test.NewClientsFromConfig(cfg, testNamespace)
+}
+
+func createServices(clients *test.Clients, count int) ([]*v1a1test.ResourceObjects, func(), error) {
+	testNames := make([]*test.ResourceNames, count)
+
+	// Initialize our service names.
+	for i := 0; i < count; i++ {
+		testNames[i] = &test.ResourceNames{
+			Service: test.AppendRandomString(fmt.Sprintf("%s-%02d", serviceName, i)),
+			// The crd.go helpers will convert to the actual image path.
+			Image: helloWorldImage,
+		}
+	}
+
+	cleanupNames := func() {
+		for i := 0; i < count; i++ {
+			test.TearDown(clients, *testNames[i])
+		}
+	}
+
+	objs := make([]*v1a1test.ResourceObjects, count)
+	begin := time.Now()
+	sos := []ktest.ServiceOption{
+		// We set a small resource alloc so that we can pack more pods into the cluster.
+		ktest.WithResourceRequirements(corev1.ResourceRequirements{
+			Limits: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("50m"),
+				corev1.ResourceMemory: resource.MustParse("50Mi"),
+			},
+			Requests: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("10m"),
+				corev1.ResourceMemory: resource.MustParse("20Mi"),
+			},
+		}),
+		ktest.WithConfigAnnotations(map[string]string{
+			autoscaling.WindowAnnotationKey: "7s",
+		}),
+	}
+	g := errgroup.Group{}
+	for i := 0; i < count; i++ {
+		ndx := i
+		g.Go(func() error {
+			var err error
+			if objs[ndx], _, err = v1a1test.CreateRunLatestServiceReady(&testing.T{}, clients, testNames[ndx], false, sos...); err != nil {
+				return fmt.Errorf("%02d: failed to create Ready service: %v", ndx, err)
+			}
+			return nil
+		})
+	}
+	if err := g.Wait(); err != nil {
+		return nil, nil, err
+	}
+	log.Printf("Created all the services in %v", time.Since(begin))
+	return objs, cleanupNames, nil
+}
+
+func parallelScaleFromZero(ctx context.Context, clients *test.Clients, objs []*v1a1test.ResourceObjects, count int, q *quickstore.Quickstore) {
+	var wg sync.WaitGroup
+	for i := 0; i < count; i++ {
+		ndx := i
+		wg.Add(1)
```

Comment:
you can just `wg.Add(count)` outside of the loop.

Reply:
Done.